### ISSUE TITLE
FNX-4956 ⁃ [taskcluster] Dep-sign APKs for Raptor consumption on each master commit

### ISSUE THREAD

**Issue by mitchhentges** [mozilla-mobile/fenix#1321]
We will have performance tests running on `fenix` in the future. One of the requirements is to have builds signed with our releng-mobile "dep" key. The "dep" ("dependency") key is very similar to our release key that's used for production/beta/nightly, except that it's used strictly for internal automation (such as for performance-testing/raptor :smile:)

This ticket requires that, on every commit to the master branch, the "release raptor" variants should dep-signed.

**Comment by ncalexan** [comment]
> We will have performance tests running on `fenix` in the future. One of the requirements is to have builds signed with the dep key.

What is the "dep" key?

> This ticket requires that, on every commit to the master branch, all the debug variants should be dep-signed.

This seems very counter-intuitive.  In general, I would much prefer to have automation Just Work on release builds -- especially for browsers, where we legitimately want to support other people automating things.

I have a plan that runs through [Bug 1533385](https://bugzilla.mozilla.org/show_bug.cgi?id=1533385) -- sadly, just backed out -- to make GeckoView itself handle this type of configuration for automation, rather than requiring vehicles (like Fenix) pass things through.  If we did this we could run Raptor directly against Fenix, and not do things that are surprising to me, like signing debug builds with what I expect is a release key.

@mitchhentges, what's the rest of the context here?

**Comment by mitchhentges** [comment]
Good points Nick! I haven't kept this ticket up-to-date, unfortunately.
* Within Releng Mobile, we have three types of keys: release keys (used for production, beta and nightlies), dep keys (so-called "dependency keys" used for automation - such as logic tests, performance tests, or perhaps staging builds) and dev keys ("development keys" used for debugging automation by hand)
* When this ticket was created, there was an issue that stopped release builds from being raptor-able (which I should've linked :grimacing:). However, the underlying issue has been resolved since this ticket was created, and the [PR that I'm preparing to land](https://github.com/mozilla-mobile/fenix/pull/1417) is indeed using release builds.

TL;DR: The "dep" key isn't the same key as used for releasing Fenix to users, and we are indeed using release builds :)

Let me know if that clears up the misinformation in my original ticket :smile:
